### PR TITLE
fix Issue 23411 - ImportC: undefined identifier __builtin_nanf

### PR DESCRIPTION
**Pull request by WalterBright**
D templates again bring C extensions to ImportC!

### CODE REVIEW

**Comment by thewilsonator** [comment]
The argument appears to determine what type of nan you get, do we care about that?

**Comment by WalterBright** [comment]
You're quite right. The no-argument case is nearly all the use cases. D currently has no mechanism for the argument case, and so far nobody has asked for it.

It's one of those things where we can consider implementing it if a need arises, like a lot of the weird C extensions floating around.

So no, we don't care about it.

**Review by dkorpel — APPROVED** [review]

File: druntime/src/__builtins.di [hunk 1]
```diff
@@ -53,6 +53,10 @@ version (CRuntime_Microsoft)
 
 version (DigitalMars)
 {
+    immutable float __nan = float.nan;
+
+    float __builtin_nanf()(char*)  { return float.nan; }
```
Comment (thewilsonator) [inline druntime/src/__builtins.di:58]: in which case this should probably be 
`float __builtin_nanf()(const char* p) { assert (!p); return float.nan; }`

**Review by RazvanN7 — APPROVED** [review]

File: druntime/src/__builtins.di (hunk 1 above)
Comment (WalterBright) [reply druntime/src/__builtins.di:58]: The use cases I've seen used a non-null string to specify an ordinary nan. So the assert would always trip.

**Pull request merged**